core/Test/Tasty/Core.hs

```diff
@@ -65,8 +68,21 @@ data Result = Result
     -- @FAIL@.
   , resultTime :: Time
     -- ^ How long it took to run the test, in seconds.
+  , printResultDetails :: Maybe ResultDetailsPrinter
+    -- ^ Optional detailed result printing action.
+    --
+    -- 'printDesultDetails' is used by providers that need to have fine grained
+    -- control about the printed results. Especially when it comes to colorization.
   }
-  deriving Show
+
+instance Show Result where
+  show result
+    =  "Result {"
```

Comment:
Still have to check before upstreaming if this `<>` and `shows` usage is efficient.